src/event.rs

```diff
@@ -3,3 +3,96 @@
 pub mod cmap;
 pub mod command;
 pub mod sdam;
+
+use std::sync::Arc;
+
+use futures_core::future::BoxFuture;
+
+use crate::event::command::CommandEvent;
+
+/// A destination for events.  Allows implicit conversion via [`From`] for concrete types for
+/// convenience with [`crate::options::ClientOptions`] construction: ```rust
+/// # use mongodb::options::ClientOptions;
+/// # fn example() {
+/// let (tx, rx) = tokio::sync::mpsc::channel(100);
+/// let options = ClientOptions::builder()
+///                 .command_event_handler(tx)
+///                 .build();
+/// # }
+/// ```
+/// or explicit construction for `Fn` traits:
+/// ```rust
+/// # use mongodb::options::ClientOptions;
+/// # use mongodb::event::EventHandler;
+/// # fn example() {
+/// let options = ClientOptions::builder()
+///                 .command_event_handler(EventHandler::callback(|ev| println!("{:?}", ev)))
+///                 .build();
+/// # }
+#[derive(Clone)]
+#[non_exhaustive]
+pub enum EventHandler<T> {
+    /// A callback.
+    Callback(Arc<dyn Fn(T) + Sync + Send>),
+    /// An async callback.
+    AsyncCallback(Arc<dyn Fn(T) -> BoxFuture<'static, ()> + Sync + Send>),
+    /// A `tokio` channel sender.
+    TokioMpsc(tokio::sync::mpsc::Sender<T>),
+}
+
+impl<T> std::fmt::Debug for EventHandler<T> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("EventHandler").finish()
+    }
+}
+
+impl<T> From<tokio::sync::mpsc::Sender<T>> for EventHandler<T> {
+    fn from(value: tokio::sync::mpsc::Sender<T>) -> Self {
+        Self::TokioMpsc(value)
+    }
+}
+
+#[allow(deprecated)]
+impl<T: crate::event::command::CommandEventHandler + 'static> From<Arc<T>>
```

Comment:
This seems reasonable to me; this could turn into a pretty major refactor for some users. What is the plan for testing the `Sender` impls in the meantime? I do think we should try to get at least coverage from the monitoring spec tests since we'll be directing users towards the new API from now on

Reply:
Good thought.  Updated the unified test runner to use a channel for command event accumulation.